### PR TITLE
Fix setup_all script

### PR DESCRIPTION
**Pull request by alinbalutoiu**
Some of the latest changes are not reflected
in setup_all.sh script causing it to fail.

This commit updates the script to ensure it can
deploy a full environment without errors or
user intervention.

### CODE REVIEW

File: setup_all.sh [hunk 1]
```diff
@@ -4,21 +4,31 @@ source config_vars.sh
 
 set -e
 
+ls config/ | grep default | while read cfgfilename; do
```
Comment (mircealungu) [inline setup_all.sh:7]: beautiful :)

File: setup_all.sh [hunk 2]
```diff
@@ -4,21 +4,31 @@ source config_vars.sh
...
+    new_cfgfilename=${cfgfilename::-8}
+    cp -n config/$cfgfilename config/$new_cfgfilename
+done
+
 # Start MySQL server
 ./run_mysql.sh
 
 # Build and run the API
-docker build -t zeeguu-api-core -f docker-files/zeeguu-api-core/Dockerfile .
+docker build --build-arg API_VERSION="1.0" -t zeeguu-api-core -f docker-files/zeeguu-api-core/Dockerfile .
```
Comment (mircealungu) [inline setup_all.sh]: @alinbalutoiu : can't we go with API_VERSION=\`cat .git/modules/Zeeguu-API/HEAD]\` here instead?

File: setup_all.sh [hunk 3]
```diff
@@ -4,21 +4,31 @@ source config_vars.sh
...
 docker run --net=host -d \
     -e MICROSOFT_TRANSLATE_API_KEY="$MICROSOFT_TRANSLATE_API_KEY" \
     -e GOOGLE_TRANSLATE_API_KEY="$GOOGLE_TRANSLATE_API_KEY" \
     -e WORDNIK_API_KEY="$WORDNIK_API_KEY" \
     --name=zeeguu-api-core zeeguu-api-core
 
+cp -n docker-files/zeeguu-web/apache-zeeguu.conf.default docker-files/zeeguu-web/apache-zeeguu.conf
```
Comment (mircealungu) [inline setup_all.sh:26]: can't we do the same thing as we did with the other config files? only if `apache-zeeguu.conf` does not exists then copy the default over it?

Comment (mircealungu) [reply setup_all.sh:26]: that would allow one to modify the configuration files, then run setup.py and have the entire infra running

**Review by mircealungu — COMMENTED** [review]
looks good. have a look at the observations pls./

File: setup_all.sh (hunk 3 above)
Comment (alinbalutoiu) [reply setup_all.sh:26]: That's why I'm passing the `-n` parameter. It should copy the file only if it doesn't exist:
```
-n, --no-clobber
              do not overwrite an existing file (overrides a previous -i
              option)
```

File: setup_all.sh (hunk 2 above)
Comment (alinbalutoiu) [reply setup_all.sh]: I'm not sure what value should API_VERSION have, but the command you suggested returns: `ref: refs/heads/abtest`.
I've changed API_VERSION to be the hash of the last commit of Zeeguu-API.

File: setup_all.sh (hunk 3 above)
Comment (mircealungu) [reply setup_all.sh:26]: ah, neat!

File: setup_all.sh (hunk 2 above)
Comment (mircealungu) [reply setup_all.sh]: oki, i like your solution

**Pull request merged**